pkg/statsd/flusher.go

```diff
@@ -20,25 +20,23 @@ type MetricFlusher struct {
 	lastFlush      int64 // Last time the metrics where aggregated. Unix timestamp in nsec.
 	lastFlushError int64 // Time of the last flush error. Unix timestamp in nsec.
 
-	flushInterval time.Duration // How often to flush metrics to the sender
-	dispatcher    Dispatcher
-	handler       Handler
-	backends      []gostatsd.Backend
-	selfIP        gostatsd.IP
-	hostname      string
-	statser       statser.Statser
+	flushInterval  time.Duration // How often to flush metrics to the sender
+	backendHandler *BackendHandler
```

Comment:
Make this an interface?

Reply:
Flusher is intrinsically tied to aggregators and the backend.  When clustering on the frontend comes in, for example, it won't have a flusher (or if it does for internal metrics, it will still be tied to the BackendHandler only).

If this has a need for being handled differently in future, it can always be put in an interface then.

Reply:
Sure. I was more thinking about unit tests.

Reply:
Ooh, gotcha.  Shall have a think on that one.